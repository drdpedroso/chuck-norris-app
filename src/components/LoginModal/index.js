import React from 'react'
import styled from 'styled-components'
import PropTypes from 'prop-types'
import { validatePassword } from 'services/utils/utils'
import { setCookie } from 'services/storage/storage'

const Title = styled.div`
  font-size: 22px;
  text-align: center;
  margin: auto;
`

const ModalWrapper = styled.div`
  height: 450px;
  width: 400px;
  position: fixed;
  left: 38%;
  background-color: white;
  border-style: solid;
  border-width: medium;
`

const Button = styled.button`
`

const Input = styled.input`
`

class LoginModal extends React.PureComponent {
  constructor() {
    super()

    this.onSubmit = this.onSubmit.bind(this)
  }
  onChange({ value }, key) {
    this.setState({
      [key]: value,
    })
  }

  onSubmit(e) {
    e.preventDefault()
    if (validatePassword(this.state.password)) {
      setCookie('logged', true)
      this.props.onSubmit()
    } else {
      alert('Password invalid')
    }
  }

  render() {
    return (
      <ModalWrapper>
        <Title>Login</Title>
        <form>
          <div>
            <Input type="text" onChange={({ target }) => this.onChange(target, 'user')} placeholder="User" />
          </div>
          <div>
            <Input type="password" onChange={({ target }) => this.onChange(target, 'password')} placeholder="Password" />
          </div>
          <Button onClick={this.onSubmit}>SUBMIT</Button>
        </form>
      </ModalWrapper>
    )
  }
}

LoginModal.propTypes = {
  onSubmit: PropTypes.func,
}

LoginModal.defaultProps = {
  onSubmit: () => {},
}

export default LoginModal
