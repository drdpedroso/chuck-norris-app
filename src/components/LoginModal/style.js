import styled from 'styled-components'

export const Title = styled.div`
  font-size: 22px;
  text-align: center;
  margin: 5px auto;
`

export const ModalWrapper = styled.div`
  height: 450px;
  width: 400px;
  position: fixed;
  left: 38%;
  background-color: white;
  border-style: solid;
  text-align:center;
  border-radius: 8px;
`

export const Button = styled.button`
  font-size: 19px
  border-style: solid;
  border-radius: 10px;
  margin-top: 8px;
`

export const Input = styled.input`
  border-radius: 4px;
  width: 80%;
  padding: 10px;
  margin: 5px auto;
`
