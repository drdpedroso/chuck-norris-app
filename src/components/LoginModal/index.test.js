import React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import LoginModal from './index'

global.alert = jest.fn()

describe('LoginModal', () => {
  it('should render correctly', () => {
    const component = renderer.create(<LoginModal />)
    expect(component).toMatchSnapshot()
  })
  it('should change state value based on key/val parameters', () => {
    const component = shallow(<LoginModal />)
    component.instance().onChange({ value: 10 }, 'test')
    expect(component.state().test).toEqual(10)
  })
  it('should trigger submit prop event if password is valid', () => {
    const password = 'abcpassword'
    const e = {
      preventDefault: jest.fn(),
    }
    const onSubmit = jest.fn()
    const component = shallow(<LoginModal onSubmit={onSubmit} />)
    component.setState({ password })
    component.instance().onSubmit(e)
    expect(e.preventDefault).toHaveBeenCalled()
    expect(onSubmit).toHaveBeenCalled()
  })
  it('should not trigger submit prop event if password is invalid', () => {
    const password = 'ab12password'
    const e = {
      preventDefault: jest.fn(),
    }
    const onSubmit = jest.fn()
    const component = shallow(<LoginModal onSubmit={onSubmit} />)
    component.setState({ password })
    component.instance().onSubmit(e)
    expect(e.preventDefault).toHaveBeenCalled()
    expect(global.alert).toHaveBeenCalled()
    expect(onSubmit).not.toHaveBeenCalled()
  })
})
