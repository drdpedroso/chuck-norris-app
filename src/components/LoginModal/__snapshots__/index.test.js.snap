// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginModal should render correctly 1`] = `
<div
  className="sc-bwzfXH ikDjqk"
>
  <div
    className="sc-bdVaJa gwBqPQ"
  >
    Login
  </div>
  <form>
    <div>
      <input
        className="sc-bxivhb eBNJKg"
        onChange={[Function]}
        placeholder="User"
        type="text"
      />
    </div>
    <div>
      <input
        className="sc-bxivhb eBNJKg"
        onChange={[Function]}
        placeholder="Password"
        type="password"
      />
    </div>
    <button
      className="sc-htpNat hHtpcc"
      onClick={[Function]}
    >
      SUBMIT
    </button>
  </form>
</div>
`;
